sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -289,10 +281,18 @@ public void addEvent(String name, Map<String, Object> attributes, OffsetDateTime
 
     private void end(int statusCode, Throwable throwable, Context context) {
         if (throwable != null) {
-            tracer.setAttribute(TracerProvider.ERROR_MSG, throwable.getMessage(), context);
-            tracer.setAttribute(TracerProvider.ERROR_TYPE, throwable.getClass().getName(), context);
+            if (statusCode == HttpConstants.StatusCodes.NOTFOUND) {
```

Comment:
Please only remove the callstack etc. for 404 with Substatusode 0 - for any SubStatusCode != 0 (like ReadSessionNotAvailable 1002 etc.) we should leave the callstack

Reply:
Yea  sure will add substatus check as well.

Reply:
Done, added 404/0 check